system/Database/SQLSRV/Builder.php

```diff
@@ -220,11 +242,54 @@ protected function _update(string $table, array $values): string
         $fullTableName = $this->getFullName($table);
 
         $statement = 'UPDATE ' . (empty($this->QBLimit) ? '' : 'TOP(' . $this->QBLimit . ') ') . $fullTableName . ' SET '
```

Comment:
Can we use `sprintf` for creating the SQL statements? With the current way, it is very error prone and quite hard to understand the concatenations happening.

Reply:
Something like this?
```php
if(empty($this->QBLimit)){
  $statement = sprintf('UPDATE %s SET',$fullTableName);
}else{
  $statement = sprintf('UPDATE TOP(%S)  %s SET',$this->QBLimit, $fullTableName);
}
```

Reply:
You can have it like this:
```php
$statement = sprintf(
    'UPDATE %s%s SET',
    empty($this->QBLimit) ? '' : 'TOP('.$this->QBLimit.') ',
    $fullTableName,
);
```